src/app-layer-ssl.c

```diff
@@ -1516,21 +1520,28 @@ static inline int TLSDecodeHSHelloExtensions(SSLState *ssl_state,
 
 end:
     if (ja3) {
-        rc = Ja3BufferAppendBuffer(&ssl_state->curr_connp->ja3_str,
-                                   &ja3_extensions);
-        if (rc == -1)
-            goto error;
-
-        if (ssl_state->current_flags & SSL_AL_FLAG_STATE_CLIENT_HELLO) {
-            rc = Ja3BufferAppendBuffer(&ssl_state->curr_connp->ja3_str,
-                                       &ja3_elliptic_curves);
+        if (ssl_state->curr_connp->ja3_hash == NULL) {
+            rc = Ja3BufferAppendBuffer(&ssl_state->curr_connp->ja3_str, &ja3_extensions);
             if (rc == -1)
                 goto error;
 
-            rc = Ja3BufferAppendBuffer(&ssl_state->curr_connp->ja3_str,
-                                       &ja3_elliptic_curves_pf);
-            if (rc == -1)
-                goto error;
+            if (ssl_state->current_flags & SSL_AL_FLAG_STATE_CLIENT_HELLO) {
+                rc = Ja3BufferAppendBuffer(&ssl_state->curr_connp->ja3_str, &ja3_elliptic_curves);
+                if (rc == -1)
+                    goto error;
+
+                rc = Ja3BufferAppendBuffer(
+                        &ssl_state->curr_connp->ja3_str, &ja3_elliptic_curves_pf);
+                if (rc == -1)
+                    goto error;
+            }
+        } else {
```

Comment:
can we skip doing ja3 in this case altogether? Like by not setting `ja3` to true at the start of the func?

Reply:
same, indeed nice